### ISSUE TITLE
[FR] Provide load arguments to model.from_pretrained when calling mlflow.transformers.load_model

### ISSUE THREAD

**Issue by jannesscholz** [mlflow/mlflow#13304]
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

I finetuned a Llama 3.1 8B on a 6-class-classification task (loaded with AutoModelForSequenceClassification.from_pretrained("meta-llama/Meta-Llama-3.1-8B", num_labels = 6) and then wrapped to a LoraModel and PeftModelForSequenceClassification). Afterwards it was saved to a mlflow server via mlflow.transformers.log_model. When trying to load the model with mlflow.transformers.load_model, a mismatch error occured since mlflow loads the default AutoModelForSequenceClassification without any parameters and therefore num_labels is set to 2. This motivates another argument for mlflow.transformers.load_model load_kwargs = {'num_labels': 6} that gives the arguments to _load_model in mlflow/transformers/model_io.py. Also, __init__.py has to be changed to provide the argument dictionary. Mlflow version: 2.16.2

### Motivation

> #### What is the use case for this feature?
Give arguments to from_pretrained when using mlflow.transformers.load_model.
> #### Why is this use case valuable to support for MLflow users in general?
It restricts wrapped models in general when loading them from mlflow.
> #### Why is it currently difficult to achieve this use case?
No arguments can be given and there is no other option to get the right amount of labels while loading the model.

### Details

I would simply add an argument to mlflow.transformers.load_model which is a dictionary that contains additional arguments for the from_pretrained method inside model_io.py. This parameter has to be accepted in __init__.py and then provided to load_model_and_components_from_huggingface_hub and load_model_and_components_from_local in model_io.py.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@jannesscholz Thanks for the feature request. Can you provide the code that doesn't work currently, but would work if the proposed feature was implemented?

**Comment by jannesscholz** [comment]
First, finetuning a LLama 3.1 like this (excluded data loading, preprocessing and some imports):
```python
from transformers import AutoModelForSequenceClassification, AutoTokenizer, Trainer, TrainingArguments, DataCollatorWithPadding
from peft import LoraConfig, prepare_model_for_kbit_training, get_peft_model

model_name = 'meta-llama/Meta-Llama-3.1-8B'
model = AutoModelForSequenceClassification.from_pretrained(
    model_name,
    num_labels = 6,
    device_map = 'cuda'
)
lora_config = LoraConfig(
    r = 32, 
    lora_alpha = 32,
    target_modules = ['q_proj', 'k_proj', 'v_proj', 'o_proj'],
    lora_dropout = 0.05, 
    bias = 'none',
    task_type = 'SEQ_CLS'
)
model = prepare_model_for_kbit_training(model)
model = get_peft_model(model, lora_config)
model.train()

tokenizer = AutoTokenizer.from_pretrained(model_name, add_prefix_space=True)

training_args = TrainingArguments(
    output_dir = model_save_path,
    logging_dir = model_save_path + '_logs',
    learning_rate = 1e-4,
    per_device_train_batch_size = 4,
    per_device_eval_batch_size = 4,
    num_train_epochs = 4,
    logging_steps = 1,
    weight_decay = 0.01,
    eval_strategy = 'epoch',
    save_strategy = 'epoch',
    save_total_limit = 2,
    metric_for_best_model = 'eval_loss',
    greater_is_better = False,
    load_best_model_at_end = True
)
trainer = Trainer(
    model = model,
    args = training_args,
    train_dataset = tokenized_data['train'],
    eval_dataset = tokenized_data['val'],
    tokenizer = tokenizer,
    data_collator = collate_fn,
    compute_metrics = compute_metrics
)
train_result = trainer.train()
```
Then, I loaded the model in another notebook to log it to mlflow (I know, I could've done it during the training. Just didn't think of it earlier on and it takes too long to repeat it):
```python
model_name = "meta-llama/Meta-Llama-3.1-8B"
base_model = AutoModelForSequenceClassification.from_pretrained(model_name, num_labels=6, device_map='cpu')
model = PeftModel.from_pretrained(base_model, model_save_path)
tokenizer = AutoTokenizer.from_pretrained(model_name)

mlflow.set_experiment("LLama")
with mlflow.start_run():
    mlflow.log_params(model.peft_config)
    mlflow.transformers.log_model(
        transformers_model = {"model": model, "tokenizer": tokenizer},
        signature = signature,
        artifact_path = "model",
        task = "text-classification"
    )
```
After that, when trying to load the model like this, a shape error occurs:
```python
mlflow_model = mlflow.transformers.load_model(
    model_uri = 'runs:/d1b1b69d3aee4e8d9dae70699b04421c/model',
    device = 'cpu'
)
```
This is due to mlflow taking the standard `LlamaForSequenceClassification.from_pretrained()` method without any parameters where the `num_labels` parameter defaults to 2 but I need 6. I changed just a few lines of code to be able to give parameters to the `model_class.from_pretrained(model_name_or_path, **load_kwargs)` call inside of model_io.py:
```python
mlflow_model = mlflow.transformers.load_model(
    model_uri = 'runs:/d1b1b69d3aee4e8d9dae70699b04421c/model',
    device = 'cpu',
    load_kwargs = {'num_labels': 6}
)
```
